packages/gatsby-plugin-nprogress/src/__tests__/gatsby-browser.js

```diff
@@ -0,0 +1,40 @@
+import NProgress from "nprogress"
+import {
+  onClientEntry,
+  onRouteUpdateDelayed,
+  onRouteUpdate,
+} from "../gatsby-browser"
+
+jest.mock(`nprogress`)
+
+describe(`gatsby-plugin-nprogress`, () => {
+  describe(`onClientEntry`, () => {
+    it(`calls NProgress.configure with the correct styles`, () => {
+      const element = {}
+      const createElement = jest.spyOn(document, `createElement`)
+      const appendChild = jest.spyOn(document.head, `appendChild`)
+      createElement.mockReturnValue(element)
+      appendChild.mockReturnValue(true)
+      NProgress.configure = jest.fn()
+      onClientEntry(null, { showSpinner: false })
+      expect(NProgress.configure.mock.calls).toMatchSnapshot()
```

Comment:
Let's tweak this. I think it's probably better to just assert a few things separately, e.g.

```js
expect(NProgress.configure).toHaveBeenCalledWith({}) // defined options here, or expect.any(Object) if you just want to assert it was called
```

```
expect(element.id).toBe(`nprogress-styles `)
expect(element.innerHTML).toMatchSnapshot()
```